### PR TITLE
(FINERACT-241) Include "Add Note" to Savings Transaction

### PR DESCRIPTION
**Pull request by ippezshelby**

### CODE REVIEW

**Comment by ippezshelby** [comment]
@nazeer1100126 , Could you please review this new PR. Hope i made it as per your comments in the previous PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java [hunk 1]
```diff
@@ -254,6 +254,12 @@ public CommandProcessingResult deposit(final Long savingsId, final JsonCommand c
         boolean isRegularTransaction = true;
         final SavingsAccountTransaction deposit = this.savingsAccountDomainService.handleDeposit(account, fmt, transactionDate,
                 transactionAmount, paymentDetail, isAccountTransfer, isRegularTransaction);
+				
+		final String noteText = command.stringValueOfParameterNamed("note");
+        if (StringUtils.isNotBlank(noteText)) {
+            final Note note = Note.savingNote(account, noteText);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:260]: need to create Note object by  savingsTransactionNote

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -297,6 +303,12 @@ public CommandProcessingResult withdrawal(final Long savingsId, final JsonComman
                 isRegularTransaction, isApplyWithdrawFee, isInterestTransfer, isWithdrawBalance);
         final SavingsAccountTransaction withdrawal = this.savingsAccountDomainService.handleWithdrawal(account, fmt, transactionDate,
                 transactionAmount, paymentDetail, transactionBooleanValues);
+				
+		final String noteText = command.stringValueOfParameterNamed("note");
+        if (StringUtils.isNotBlank(noteText)) {
+            final Note note = Note.savingNote(account, noteText);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:309]: need to create Note object by  savingsTransactionNote

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionDataValidator.java [hunk 3]
```diff
@@ -94,6 +95,9 @@ public void validate(final JsonCommand command) {
 
         final BigDecimal transactionAmount = this.fromApiJsonHelper.extractBigDecimalWithLocaleNamed(transactionAmountParamName, element);
         baseDataValidator.reset().parameter(transactionAmountParamName).value(transactionAmount).notNull().positiveAmount();
+		
+		final String note = this.fromApiJsonHelper.extractStringNamed(noteParamName, element);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionDataValidator.java:99]: Note can not be mandatory data. So ignore if it is null

**Review by nazeer1100126 — CHANGES_REQUESTED** [review]
This code has saving the transaction note details. But I don't see any code related to getting this note. 
Is there any use of saving transaction note without showing in the gui? I believe this is the case with even with loan transaction notes.

**Comment by ippezshelby** [comment]
@nazeer1100126 , am working on the changes you requested for FINERACT-241. But i have a challenge of relating m_note table with either m_savings_account_transaction or m_savings_account for getting this note to show it in the UI. Any clue here?

It would be better to match each notes with its transactions so that we know how each transaction has be narrated. So we must relate m_note table with m_savings_account_transaction tr  'left join m_note mn on tr.id = mn.savings_account_transaction_id' . But duplicates each transaction row in the result-set.
This query would have worked but i realized savings_account_transaction_id is not getting inserted into m_note when a transaction occurs.
select tr.id as transactionId, sa.account_no as Account,
tr.transaction_date as transactionDate, tr.amount as transactionAmount,
tr.running_balance_derived as runningBalance,
au.username as submittedByUsername,
mn.note as note
from m_savings_account sa
join m_savings_account_transaction tr on tr.savings_account_id = sa.id
left join m_appuser au on au.id=tr.appuser_id
left join m_note mn on tr.id = mn.savings_account_transaction_id

**Comment by nazeer1100126** [comment]
@Ippezrobert If you see my comment, you need to create note object by passing savings account, savings transaction objects. In your changes you are not doing it. rest of the queries are similar to loan transaction notes. Please refer loan transaction note implementation. If you need some more details, we can have a skype or hangouts call.

**Comment by avikganguly01** [comment]
@Ippezrobert : Would you be able to make the suggested changes?

**Comment by nazeer1100126** [comment]
@myrle-krantz please close this PR. I have already implemented this feature and it is merged recently

**Comment by myrle-krantz** [comment]
Have closed.  For future reference, you can close a PR with an empty commit message as explained here:

https://cwiki.apache.org/confluence/display/FINERACT/Merge+a+pull+request+to+a+mirrored+Apache+repo

**Comment by nazeer1100126** [comment]
@myrle-krantz Thank you :-)

**Pull request closed without merging**